rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,151 @@
+that Kibana loads in the browser.
+
+This RFC proposes a Screenshot Mode Service as a low-level plugin that allows
+other plugins (UI code) to make choices when the page is rendering for a screenshot.
+
+More background on how Reporting currently works, including the lifecycle of
+creating a PNG report, is here: https://github.com/elastic/kibana/issues/59396
+
+# Motivation
+
+The Reporting team wants all applications to support a customized URLs, such as
+Canvas does with its `#/export/workpad/pdf/{workpadId}` UI route. The
+customized URL is where an app can solve any rendering issue in a PDF or PNG,
+without needing extra CSS to be injected into the page.
+
+However, many low-level plugins have been added to the UI over time. These run
+on every page and an application can not turn them off. Reporting performance
+is negatively affected by this type of code. When the Reporting team analyzes
+customer logs to figure out why a job timed out, we sometimes see requests for
+the newsfeed API and telemetry API: services that aren't needed during a
+reporting job.
+
+In 7.12.0, using the customized `/export/workpad/pdf` in Canvas, the Sample
+Data Flights workpad loads 163 requests. Most of thees requests don't come from
+the app itself but from the application container code that Canvas can't turn
+off.
+
+# Detailed design
+
+The Screenshot Mode Service is an entirely new plugin that has an API method
+that returns a Boolean. The return value tells the plugin whether or not it
+should render itself to optimize for non-interactivity.
+
+The plugin is low-level as it has no dependencies of its own, so other
+low-level plugins can depend on it.
+
+## Interface
+A plugin would depend on `screenshotMode` in kibana.json. That provides
+`screenshotMode` as a plugin object. The plugin's purpose is to know when the
+page is rendered for screenshot capture, and to interact with plugins through
+an API. It allows plugins to decides what to do with the screenshot mode
+information.
+
+```
+interface IScreenshotModeServiceSetup {
+  isScreenshotMode: () => boolean;
+}
+```
+
+The plugin knows the screenshot mode from request headers: this interface is
+constructed from a class that refers to information sent via a custom
+proprietary header:
+
+```
+interface HeaderData {
+  'X-Screenshot-Mode': true
+}
+
+class ScreenshotModeServiceSetup implements IScreenshotModeServiceSetup {
+  constructor(rawData: HeaderData) {}
+  public isScreenshotMode (): boolean {}
+}
+```
+
+The Reporting headless browser that opens the page can inject custom headers
+into the request. Teams should be able to test how their app renders when
+loaded with this header. They could use a web debugging proxy, or perhaps the
+new service should support a URL parameter which triggers screenshot mode to be
+enabled, for easier testing.
+
+# Basic example
+
+When Kibana loads initially, there is a Newsfeed plugin in the UI that
```

Comment:
For this particular example, would be even simpler if the newsfeed client-side plugin not loaded at all in screenshot mode? 
I am curious if this is possible during "the hook-into rendering process" that you've described.
_Or this could definitely be possible if screenshot thingy is part of the core and plugins can declare if they need to be skipped in screenshot mode in `kibana.json`_